lib/compiler-cranelift/src/translator/func_translator.rs

```diff
@@ -159,17 +159,17 @@ fn declare_wasm_parameters<FE: FuncEnvironment + ?Sized>(
 ///
 /// Declare local variables, starting from `num_params`.
 fn parse_local_decls<FE: FuncEnvironment + ?Sized>(
-    reader: &mut dyn FunctionBinaryReader,
+    reader: &wasmer_compiler::FunctionReader,
     builder: &mut FunctionBuilder,
     num_params: usize,
     environ: &mut FE,
 ) -> WasmResult<()> {
     let mut next_local = num_params;
-    let local_count = reader.read_local_count()?;
-
+    let mut local_reader = reader.get_locals_reader()?;
+    let local_count = local_reader.get_count();
     for _ in 0..local_count {
-        builder.set_srcloc(cur_srcloc(reader));
-        let (count, ty) = reader.read_local_decl()?;
+        builder.set_srcloc(ir::SourceLoc::new(local_reader.original_position() as u32));
```

Comment:
TODO: This is actually incorrect as it will assign all locals to the byte offset at which the locals reader started reading.

On the other hand… I'm not sure if we emit debug info in any meaningful capacity right now.